source/adios2/core/Engine.h

```diff
@@ -353,19 +365,68 @@ class Engine
      * Unallocated version, ADIOS will allocate space for incoming data
      * @param variableName
      */
-    void ScheduleRead(const std::string variableName)
-    {
-        ScheduleRead(variableName, nullptr);
-    }
+    virtual void ScheduleRead(const std::string &variableName);
 
     /**
      * Unallocated unspecified version, ADIOS will receive any variable and will
      * allocate space for incoming data
```

Comment:
I am not sure what you mean by that. The highlighted function is the one that allows the user say hey adios allocate the space for me to read this variable in.